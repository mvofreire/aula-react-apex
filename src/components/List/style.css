.list {
  list-style: none;
  padding: 0;
  margin: 0;
}

.list-item {
  margin: 10px 0px;
  display: flex;
  justify-content: space-between;
  align-items: center;
}

.list-item-label {
  margin-right: 20px;
}

.list-blank-result {
  color: #999;
  margin-top: 100px;
  display: block;
  text-align: center;
}

.list-clear-all{
  float: right;
}
