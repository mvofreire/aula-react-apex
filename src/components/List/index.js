import React from 'react'
import Button from '../Button'

import "./style.css"

function List({ tasks = [], onRemoveClick }) {
  return (
    <ul className='list'>
      {tasks.map((task, index) => (
        <li className='list-item' key={`list-item-${index}`}>
          <span className='list-item-label'>{task}</span>
          <Button label="Remover" onClick={() => {
            onRemoveClick(index)
          }} />
        </li>
      ))}
    </ul>
  )
}

export default List