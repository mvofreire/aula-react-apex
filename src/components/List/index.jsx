import React, { useCallback } from "react";
import { Button, Empty, List as AntList } from "antd";
import "./style.css";
import { Link } from "react-router-dom";

const List = ({ tasks = [], onRemoveClick, onClearAll }) => {
  const handleRemoveClick = useCallback((index, e) => {
    e.preventDefault();
    e.stopPropagation();

    onRemoveClick(index);
  }, []);

  if (tasks.length === 0) {
    return <Empty />;
  }

  return (
    <>
      <AntList>
        {tasks.map((item, i) => (
          <Link key={i} to={`/${i}`}>
            <AntList.Item
              extra={
                <Button
                  danger
                  type="primary"
                  onClick={handleRemoveClick.bind(null, i)}
                >
                  x
                </Button>
              }
            >
              {item}
            </AntList.Item>
          </Link>
        ))}
      </AntList>
      {tasks.length > 0 && (
        <Button
          className="btn-clear-all"
          danger
          type="primary"
          onClick={onClearAll}
        >
          Remove All
        </Button>
      )}
    </>
  );
};

export default List;
