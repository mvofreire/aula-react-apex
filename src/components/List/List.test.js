import { render, act, fireEvent, cleanup } from '@testing-library/react';
import List from '.';

afterEach(cleanup)

describe('Input', () => {
  const data = [
    'item 1',
    'item 2',
    'item 3',
  ]
  const renderComponent = (props) => {
    return render(<List tasks={data} {...props} />)
  }

  test('Should render in screen', () => {
    const { container } = renderComponent()
    expect(container).toBeInTheDocument()
  })

  test('Should match with snapshot', () => {
    const { container } = renderComponent()
    expect(container).toMatchSnapshot('default')
  })

  test('Should call callback on remove item', () => {
    const onRemoveClick = jest.fn();
    const { getAllByText } = renderComponent({ onRemoveClick })

    act(() => {
      fireEvent.click(getAllByText(/x/)[0])
    })

    expect(onRemoveClick).toBeCalledTimes(1)
  })

})