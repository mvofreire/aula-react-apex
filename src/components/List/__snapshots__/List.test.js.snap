// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Should match with snapshot: default 1`] = `
<div>
  <ul
    class="list"
  >
    <li
      class="list-item"
    >
      <span
        class="list-item-label"
      >
        item 1
      </span>
      <button
        class="btn secondary "
        type="button"
      >
        x
      </button>
    </li>
    <li
      class="list-item"
    >
      <span
        class="list-item-label"
      >
        item 2
      </span>
      <button
        class="btn secondary "
        type="button"
      >
        x
      </button>
    </li>
    <li
      class="list-item"
    >
      <span
        class="list-item-label"
      >
        item 3
      </span>
      <button
        class="btn secondary "
        type="button"
      >
        x
      </button>
    </li>
    <button
      class="btn secondary list-clear-all"
      type="button"
    >
      Apagar Todos
    </button>
  </ul>
</div>
`;
